CHANGELOG.md

```diff
@@ -17,6 +17,8 @@ Inspired from [Keep a Changelog](https://keepachangelog.com/en/1.0.0/)
 
 ### Fixed
 
+- Fixes empty request body on retry with compression enabled. ([#543](https://github.com/opensearch-project/opensearch-go/pull/543))
```

Comment:
Nit: remove period at the end of the sentence, other lines don't have it.

Reply:
Done.